src/loaders/coco_load.c

```diff
@@ -276,20 +277,22 @@ static int coco_load(struct module_data *m, HIO_HANDLE *f, const int start)
 		if (libxmp_alloc_pattern_tracks(mod, i, 64) < 0)
 			return -1;
 
-		for (j = 0; j < (64 * mod->chn); j++) {
-			event = &EVENT (i, j % mod->chn, j / mod->chn);
-			event->fxp = hio_read8(f);
-			event->fxt = hio_read8(f);
-			event->ins = hio_read8(f);
-			event->note = hio_read8(f);
-			if (event->note)
-				event->note += 12;
-
-			if (hio_error(f)) {
-				return -1;
+		for (j = 0; j < 64; j++) {
+			for (k = 0; k < mod->chn; k++) {
+				event = &EVENT(i, k, j);
```

Comment:
Unless I'm missing something (newly woke up), this is reading (0,0,0), (0,1,0), (0,2,0)
versus the old (0,0,0), (0,0,1), (0,0,2): are we sure that it's correct?

Reply:
That is what the old version did too: the modulo is on the second parameter and the division is on the third. (I think if those had been reversed the format test would fail entirely due to different pattern data.)

Reply:
Oh hell, as I'm still waking up..